control-plane/api-gateway/common/translation.go

```diff
@@ -358,38 +398,52 @@ func (t ResourceTranslator) translateHTTPFilters(filters []gwv1beta1.HTTPRouteFi
 				// this should never be the case because we only translate a route if it's actually valid, and if we're missing filters during the validation step, then we won't get here
 				continue
 			}
+
 			switch filter.ExtensionRef.Kind {
 			case v1alpha1.RouteRetryFilterKind:
-
-				retryFilterCRD := crdFilter.(*v1alpha1.RouteRetryFilter)
-				//new filter that needs to be appended
-
-				retryFilter = &api.RetryFilter{
-					NumRetries:            retryFilterCRD.Spec.NumRetries,
-					RetryOn:               retryFilterCRD.Spec.RetryOn,
-					RetryOnStatusCodes:    retryFilterCRD.Spec.RetryOnStatusCodes,
-					RetryOnConnectFailure: retryFilterCRD.Spec.RetryOnConnectFailure,
-				}
-
+				retryFilter = t.translateRouteRetryFilter(crdFilter.(*v1alpha1.RouteRetryFilter))
 			case v1alpha1.RouteTimeoutFilterKind:
-
-				timeoutFilterCRD := crdFilter.(*v1alpha1.RouteTimeoutFilter)
-				//new filter that needs to be appended
-
-				timeoutFilter = &api.TimeoutFilter{
-					RequestTimeout: timeoutFilterCRD.Spec.RequestTimeout,
-					IdleTimeout:    timeoutFilterCRD.Spec.IdleTimeout,
-				}
-
+				timeoutFilter = t.translateRouteTimeoutFilter(crdFilter.(*v1alpha1.RouteTimeoutFilter))
+			case v1alpha1.RouteAuthFilterKind:
+				jwtFilter = t.translateRouteJWTFilter(crdFilter.(*v1alpha1.RouteAuthFilter))
 			}
 		}
-
 	}
+
 	return api.HTTPFilters{
 		Headers:       []api.HTTPHeaderFilter{consulFilter},
 		URLRewrite:    urlRewrite,
 		RetryFilter:   retryFilter,
 		TimeoutFilter: timeoutFilter,
+		JWT:           jwtFilter,
+	}
+}
+
+func makeJWTFilter(routeJWTFilter *v1alpha1.RouteAuthFilter) *api.JWTFilter {
+	if routeJWTFilter.Spec.JWT == nil || routeJWTFilter.Spec.JWT.Providers == nil {
+		return nil
+	}
+
+	// For each provider, translate the claims to the Consul config entry format.
+	providers := make([]*api.APIGatewayJWTProvider, 0, len(routeJWTFilter.Spec.JWT.Providers))
+	for _, provider := range routeJWTFilter.Spec.JWT.Providers {
+
+		claims := make([]*api.APIGatewayJWTClaimVerification, 0, len(provider.VerifyClaims))
+		for _, claim := range provider.VerifyClaims {
+			claims = append(claims, &api.APIGatewayJWTClaimVerification{
+				Path:  claim.Path,
+				Value: claim.Value,
+			})
+		}
+
+		providers = append(providers, &api.APIGatewayJWTProvider{
+			Name:         provider.Name,
+			VerifyClaims: claims,
+		})
+	}
+
+	return &api.JWTFilter{
+		Providers: providers,
```

Comment:
To keep with some of the way a lot of this file is done functionally (in part to break down the error handling/internal nil pointer checking) and since the helpers already exist, wondering if it'd be a bit more terse/standardized to do something like:

```go
func makeJWTFilter(filter *v1alpha1.RouteAuthFilter) *api.JWTFilter {
	if filter.Spec.JWT == nil {
		return nil
	}

	return &api.JWTFilter{
		Providers: ConvertSliceFunc(filter.Spec.JWT.Providers, t.translateJWTProvider),
	})
}
```